docs-starlight/src/content/docs/05-troubleshooting/03-performance.mdx

```diff
@@ -0,0 +1,149 @@
+
+When in doubt, [measure the performance](#measuring-performance) before and after enabling the provider cache server to see if it's a net win for your use case.
+
+We are coordinating with the OpenTofu team to improve the behavior of concurrent access to provider cache, so that this flag will be unnecessary for most users in the future.
+
+See [#1939](https://github.com/opentofu/opentofu/issues/1483) for more details.
+
+### Fetching Output From State
+
+Under the hood, Terragrunt dependency blocks leverage the OpenTofu/Terraform `output -json` command to fetch outputs from one unit and leverage them in another.
+
+The OpenTofu/Terraform `output -json` command does a bit more work than simply fetching output values from state, and a significant portion of that slowdown is loading providers, which it doesn't really need in most cases.
+
+You can significantly improve the performance of dependency blocks by using the `--dependency-fetch-output-from-state` flag. When the flag is set, Terragrunt will directly fetch the backend state file from S3 and parse it directly, avoiding any overhead incurred by calling the `output -json` command.
+
+For example:
+
+```shell
+terragrunt run --all plan --dependency-fetch-output-from-state
+```
+
+#### Fetching Output From State - Gotchas
+
+The first thing you need to be aware of when considering usage of the `--dependency-fetch-output-from-state` flag is that it only works for S3 backends. If you are using a different backend, you should not use this flag.
+
+Next, you should be aware that there is no guarantee that OpenTofu/Terraform will maintain the existing schema of their state files, so there is also no guarantee that the flag will work as expected in future versions of OpenTofu/Terraform.
+
+We are coordinating with the OpenTofu team to improve the performance of the `output` command, and we hope that this flag will be unnecessary for most users in the future.
+
+See [#1549](https://github.com/opentofu/opentofu/issues/1549) for more details.
+
+### Skip Dependency Inputs
+
+A rarely used capability of Terragrunt dependency blocks is the ability to read inputs from dependencies. There's very little in the way of valid use-cases for this, and supporting this behavior is a source of significant performance penalty for Terragrunt users.
+
+The reason for this is that Terragrunt needs to recursively read inputs from dependencies, and their dependencies, all the way down to the ancestral Terragrunt unit they all have in common to be able to expose the inputs to the dependent unit.
+
+You can mitigate this performance penalty by using the `skip-dependency-inputs` strict control.
+
+Enabling this strict control will enable a breaking change in Terragrunt behavior, making it so that dependency blocks no longer read inputs from dependencies.
+
+In the future, we will be removing this capability from Terragrunt, so you won't need to use this strict control.
```

Comment:
Do we know when the removal will be? 1.0?  Can we say that explicitly?

Reply:
It's on the road to 1.0, and there's no deadline at the moment.